Dn02/src/gauss_legendre.jl

```diff
@@ -25,13 +25,94 @@ const GL10_weights = [
     0.0666713443086881
 ]
 
+const GL12_nodes = [
+    -0.1252334085114689,
+     0.1252334085114689,
+    -0.3678314989981802,
+     0.3678314989981802,
+    -0.5873179542866175,
+     0.5873179542866175,
+    -0.7699026741943047,
+     0.7699026741943047,
+    -0.9041172563704749,
+     0.9041172563704749,
+    -0.9815606342467192,
+     0.9815606342467192
+]
+
+const GL12_weights = [
+    0.2491470458134028,
+    0.2491470458134028,
+    0.2334925365383548,
+    0.2334925365383548,
+    0.2031674267230659,
+    0.2031674267230659,
+    0.1600783285433462,
+    0.1600783285433462,
+    0.1069393259953184,
+    0.1069393259953184,
+    0.0471753363865118,
+    0.0471753363865118
+]
+
+const GL14_nodes = [
+    -0.1080549487073437,
+     0.1080549487073437,
+    -0.3191123689278897,
+     0.3191123689278897,
+    -0.5152486363581541,
+     0.5152486363581541,
+    -0.6872929048116855,
+     0.6872929048116855,
+    -0.8272013150697650,
+     0.8272013150697650,
+    -0.9284348836635735,
+     0.9284348836635735,
+    -0.9862838086968123,
+     0.9862838086968123
+]
+
+const GL14_weights = [
+    0.2152638534631578,
+    0.2152638534631578,
+    0.2051984637212956,
+    0.2051984637212956,
+    0.1855383974779378,
+    0.1855383974779378,
+    0.1572031671581935,
+    0.1572031671581935,
+    0.1215185706879032,
+    0.1215185706879032,
+    0.0801580871597602,
+    0.0801580871597602,
+    0.0351194603317519,
+    0.0351194603317519
+]
+
+# store all supported rules in a dictionary
+const GL_rules = Dict(
+    10 => (GL10_nodes, GL10_weights),
+    12 => (GL12_nodes, GL12_weights),
+    14 => (GL14_nodes, GL14_weights)
+)
+
 function cdf_legendre(x::Float64; n::Int=10)
-    # Map [0, x] -> [-1,1]
+    # check if rule exists
+    haskey(GL_rules, n) || error("Gauss–Legendre rule for n=$n not available. Choose from: $(collect(keys(GL_rules)))")
+
+    nodes, weights = GL_rules[n]
+
+    # symmetry: reduce to positive x
+    if x < 0
```

Comment:
Due to the properties of the function ($\lim_{x \to -\infty} = 0$) it is better to do it the other way around: use symmetry to calculate positive values. This way you avoid catastrophic subtraction $1-F(-x)$ where $F(-x) \approx 1$.